Core.ApplicationServices/UserService.cs

```diff
@@ -306,31 +306,58 @@ public void Dispose()
                 .Transform(Result<IQueryable<User>, OperationError>.Success);
         }
 
-        public Maybe<OperationError> DeleteUserFromKitos(Guid userUuid)
+        public Maybe<OperationError> InitiateUserDeletion(Guid userUuid, int? scopedToOrganizationId = null)
         {
-            using var transaction = _transactionManager.Begin();
-
             var user = _userRepository.AsQueryable().ByUuid(userUuid);
             if (user == null)
                 return new OperationError(OperationFailure.NotFound);
             if(_organizationalUserContext.UserId == user.Id)
                 return new OperationError("You cannot delete a user you are currently logged in as", OperationFailure.Forbidden);
 
+            if (scopedToOrganizationId.HasValue && scopedToOrganizationId.Value != 0)
+                return DeleteUserFromOrganizationOrKitos(scopedToOrganizationId.Value, user);
+            
+            return _authorizationContext.AllowDelete(user) 
```

Comment:
the authorization can be moved to the deleteuserfromkitos.. right?.. then you check that in the beginning of the method (the allowdelete) and this line will just be "return DeleteUserFromKitos"